### ISSUE TITLE
Which model is being used in the backend [User privacy] ? 

### ISSUE THREAD

**Issue by ravindra-ut** [gorilla-llm/gorilla-cli#1]
Hi,

Gorilla-CLI fuses the capabilities of various Language Learning Models (LLMs) like [Gorilla LLM](https://github.com/ShishirPatil/gorilla/), OpenAI's GPT-4, Claude v1, and others to present a user-friendly command-line interface.

Does it mean user queries are submitted to GPT4(OpenAI) or Claudev1(Anthropic) ?
Can I select or opt for Gorilla only inference ?
Can user be provided option with setting up Gorilla model on local setup ? Is there plan to release Gorilla model checkpoints ?

**Comment by ShishirPatil** [comment]
Thanks for trying out Gorilla-cli @ravindra-ut . This is a great suggestion. I was wondering perhaps user can pass an argument `--model` and include that in the RESTful call? Do you want to maybe raise a PR for this - if you are interested  - would love to have you join this project as a contributor! I can modify the back end to respect the model choice.

**Comment by ShishirPatil** [comment]
Yup always! The models are evolving rapidly rite now. So, we will release the model checkpoints once it stabilizes to a certain extent.

**Comment by Noppapon** [comment]
@ShishirPatil An issue that may arise when passing the `--model` argument is that some users may find it cumbersome to specify it with every command they run. I think it would be great to have a `config.json` file that saves user settings, and we can potentially remove `USER_ID_FILE` and `UPDATE_CHECK_FILE` as well. All the information could then be stored within the `config.json` file. The file will also be useful and make the code cleaner as the system grows and has more complex features that require individual user information to be saved. 

With this approach, we would check the `config.json` if `models` are specified, and if so, pass it into the RESTFUL call.

**Comment by ShishirPatil** [comment]
Hey @Noppapon thanks for taking this on! Yeah, that is a good suggestion - we can have a `config.json` that can have all this information in one place. Maybe we can still allow the users to over-ride the config with the `--model` or `-m` flag, in those cases where they want to use a model one-off? Thoughts!
Otherwise, this sounds good - let me know when you raise the PR and I can review it. Thanks man!

**Comment by Noppapon** [comment]
Thank you, @ShishirPatil ! I've made several additions in the same PR to clean up the code related to argument passing and the UI, using the Click framework. Along with that, users can now execute essential commands like `--help`, `--version.`, and some more, which I think would be useful. The PR is now ready for review.
